syntax="proto2";
package fcgiprotocol;

enum FCGI_Header{
    BEGIN_REQUEST = 1;
    ABORT_REQUEST = 2;
    END_REQUEST = 3;
    PARAMS = 4;
    STDIN = 5;
    STDOUT = 6;
    STDERR = 7;
    DATA = 8;
    GET_VALUES = 9;
    GET_VALUES_RESULT = 10;
    UNKNOWN_TYPE = 11;
}

enum Header_Const{
    NULL_REQUEST_ID = 0;
    VERSION_1 = 1;
    HEADER_LEN = 8;
}
enum Conn_Status{
    KEEP_CONN = 1;
}

enum FCGI_Role{
    RESPONDER = 1;
    AUTHORIZER = 2;
    FILTER = 3;
}

enum Protocol_Status{
    REQUEST_COMPLETE = 0;
    CANT_MPX_CONN = 1;
    OVERLOADED = 2;
    UNKNOWN_ROLE = 3;
}

message BeginRequest{
    optional int32 role=1;
    optional int32 flags=2;
    optional string reserved1=3;

}

message MyPageContent {
    required string key = 1;
    required string value = 2;
}
message Params {
    repeated MyPageContent paramvalues= 1;
}

message Record{
    optional int32 version = 1;
    optional int32 type = 2;
    required int32 request_id = 3;
    optional int32 reserved = 4;
    optional int32 contentLength = 5;
    optional int32 paddingLength = 6;
    optional string contentData = 7;
    optional string paddingData = 8;
    
}

message Stdin{
    optional string contentData=1;
    optional int32 type=2;
}